File: data/Models/Skill.js
import mongoose from 'mongoose';

const skillSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    unique: true,
    trim: true,
  },
  category: {
    type: String,
    required: true,
    enum: ['Frontend', 'Backend', 'Data', 'Design', 'DevOps'],
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
});

export const Skill = mongoose.model('Skill', skillSchema);

File: Models/Skill.cs
using System;
using System.Collections.Generic;

namespace TeamWeave.Models
{
    public class Skill
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Category { get; set; } // "Frontend", "Backend", "Data", "Design", etc.
        public DateTime CreatedAt { get; set; }

        public virtual ICollection<ParticipantSkill> ParticipantSkills { get; set; }

        public Skill()
        {
            ParticipantSkills = new HashSet<ParticipantSkill>();
            CreatedAt = DateTime.Now;
        }
    }
}
